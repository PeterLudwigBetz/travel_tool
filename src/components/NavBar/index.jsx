import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import travela from '../../images/travela.svg';
import icon from '../../images/drop-down-icon.svg';
import notification from '../../images/notification.svg';
import SearchBar from '../SearchBar';
import Button from '../Button';
import ImageLink from '../ImageLInk';
import './NavBar.scss';

/**
 * @description - Contains SearhBar componentents, Logo, user avatar and logout
 *
 * @class NavBar
 *
 * @extends {PureComponent}
 *
 */

class NavBar extends PureComponent {
  renderLogo() {
    return (
      <span className="navbar__logo-icons">
        <img src={travela} alt="Andela Logo" />
      </span>
    );
  }

  renderNotification() {
    const { onNotificationToggle } = this.props;
    return (
      <div
        id="notification"
        onClick={onNotificationToggle}
        className="navbar__nav-size"
        role="presentation"
      >
        <span className="material-icons mdl-badge navbar__badge" data-badge="12">
          <img
            src={notification}
            alt="Notification"
            className="navbar__navbar-notification"
          />
        </span>
      </div>
    );
  }

  renderUserIcons() {
    const { avatar } = this.props;
    return (
      <div>
        <span className="navbar__mdl-icons">
          <ImageLink imageSrc={avatar} altText="Andela Logo" imageClass="navbar__mdl-upic" />
          <span className="navbar__text-size">
            Silm Momoh
          </span>
        </span>
        <span>
          <Button 
            imageSrc={icon} altText="Dropdown Icon" buttonId="demo-menu-lower-right" imageClass="navbar__mdl-Icon" buttonType="button" 
            buttonClass="mdl-button mdl-js-button mdl-button--icon mdl-Icons" />
          <div className="navbar__mdl-list">
            <ul htmlFor="demo-menu-lower-right" className="mdl-menu mdl-menu--bottom-right mdl-js-menu mdl-js-ripple-effect navbar__menu">
              <li className="mdl-menu__item navbar__menu-item">
                Logout
              </li>
            </ul>
          </div>
        </span>
      </div>
    );
  }

  render() {
    return (
      <header className="mdl-layout__header navbar__layout_header">
        <div className="mdl-layout__header-row">
          <div className="navbar__nav-size navbar__logo-icons">
            {this.renderLogo()}
          </div>
          <div className="mdl-layout-spacer" />
          <div className="navbar__search-size">
            <SearchBar />
          </div>
          <nav className="mdl-navigation">
            {this.renderNotification()}
            <div className="navbar__user-icon navbar__nav-size">
              {this.renderUserIcons()}
            </div>
          </nav>
        </div>
      </header>
    );
  }
}

NavBar.propTypes = {
  avatar: PropTypes.string.isRequired,
  onNotificationToggle: PropTypes.func.isRequired
};

export default NavBar;
