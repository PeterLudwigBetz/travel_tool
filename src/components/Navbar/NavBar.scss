.mdl-layout__header {
  height: 78px;
  width: 1440px;
  background-color: #ffffff;
}

.mdl-badge.mdl-badge:after {
  box-sizing: border-box;
  top: 15px;
  right: -10px;
  height: 17px;
  width: 17px;
  border: 1px solid #ffffff;
  background-color: #0459e4;
  color: #ffffff;
  font-family: "DIN Pro";
  font-size: 10px;
  font-weight: 500;
  letter-spacing: -0.13px;
  line-height: 12px;
  text-align: center;
}

.mdl-menu__item {
  height: 21px;
  width: 50px;
  color: #3359db;
  font-family: "DIN Pro";
  font-size: 16px;
  line-height: 21px;
}
.mdl-menu {
  width: 50px;
  height: 25px;
  border-radius: 2px;
  background-color: #ffffff;
}
.navbar {
  &__logo-icons {
    height: 40px;
    width: 130px;
    margin-top: 15px;
  }

  &__nav-size {
    margin-right: 25px;
  }
  &__search-size {
    margin-right: -40px;
  }
  &__navbar-notification {
    height: 21.13px;
    width: 16.25px;
    margin-top: 15px;
    color: gray;
  }

  &__user-icon {
    height: 32px;
    width: 161px;
    margin: 15px 150px 0 0;
  }

  &__text-size {
    margin: 7px;
    color: #4f4f4f;
    font-family: "DIN Pro";
    font-size: 14px;
    height: 21px;
    width: 90px;
    line-height: 21px;
  }

  &__navbar-dropdown {
    color: gray;
  }

  &__mdl-upic {
    box-sizing: border-box;
    height: 34px;
    width: 34px;
    border: 1px solid #ffffff;
    margin-right: 2px;
  }

  &__mdl-Icons {
    margin-bottom: 5px;
  }
  &__mdl-Icon {
    height: 4.99px;
    width: 8.99px;
  }

  &__mdl-list {
    position: relative;
    top: -55px;
    left: 8px;
    margin: 0 2px;
  }
}
