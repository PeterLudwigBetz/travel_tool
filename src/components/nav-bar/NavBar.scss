.mdl-layout__header {
    height: 78px;
    width: 1440px;
    background-color: #ffffff;
    box-shadow: none;
    display: block;
    border-bottom: 1px solid #e9e9e9;
}

.mdl-layout__header-row {
    padding: 0 40px 0 51px;
}

.mdl-badge.mdl-badge:after {
    box-sizing: border-box;
    top: 15px;
    right: -10px;
    height: 17px;
    width: 17px;
    border: 1px solid #ffffff;
    background-color: #0459e4;
    color: #ffffff;
    font-family: 'DIN Pro';
    font-size: 10px;
    font-weight: 500;
    letter-spacing: -0.13px;
    line-height: 12px;
    text-align: center;
}

.navbar {
    &__layout_header {
        height: 8.5%;
        margin: auto;
        width: 100%;
        background-color: #ffffff;
        box-shadow: none;
        border-bottom: 1px solid #e9e9e9;
        @media (max-width: 839px) and (min-width: 480px) {
            width: 99%; // background-color: blue;
        }
        @media (max-width: 479px) {
            // width: auto;
            height: auto;
        }
    }
    &__search-icon {
        display: none;
        @media (max-width: 479px) {
            display: block;
            left: 0;
            color: blue;
            position: absolute;
            margin-left: 237px;
            top: 24px;
        }
        &--btn {
            display: none;
            @media (max-width: 479px) {
                display: block;
            }
        }
    }
    &__menu-item {
        height: 21px;
        color: #3359db;
        font-family: 'DIN Pro';
        font-size: 16px;
        line-height: 21px;
    }
    &__menu {
        min-width: 105px;
        height: auto;
        border-radius: 2px;
        background-color: #ffffff;
    }
    &__logo-icons {
        height: 40px;
        width: 130px;
        margin-top: 15px;
    }
    &__nav-size {
        margin-right: 25px;
        @media (max-width: 839px) and (min-width: 480px) {
            margin-right: -24px;
            margin-left: 20px;
        }
        @media (max-width: 479px) {
            margin-right: -31px;
        }
    }
    &__search-size {
        margin-right: -40px;
        @media (max-width: 839px) and (min-width: 480px) {
            margin-right: -83px;
        }
        @media (max-width: 479px) {
            display: none;
            margin-right: 0px;
        }
    }
    &__navbar-notification {
        height: 21.13px;
        width: 16.25px;
        margin-top: 15px;
        color: gray;
        cursor: pointer;
    }
    &__user-icon {
        height: 32px;
        margin: 15px 20px 0 0;
    }
    &__text-size {
        margin: 7px;
        color: #4f4f4f;
        font-family: 'DIN Pro';
        font-size: 14px;
        height: 21px;
        width: 90px;
        line-height: 21px;
    }
    &__navbar-dropdown {
        color: gray;
    }
    &__mdl-upic {
        box-sizing: border-box;
        height: 34px;
        width: 34px;
        border: 1px solid #ffffff;
        margin-right: 2px;
        border-radius: 50%;
}
    &__mdl-Icons {
        margin-bottom: 5px;
    }
    &__mdl-Icon {
        height: 4.99px;
        width: 8.99px;
    }
    &__mdl-list {
        position: relative;
        top: -53px;
        left: -11px;
        margin: 0 1px;
    }
}