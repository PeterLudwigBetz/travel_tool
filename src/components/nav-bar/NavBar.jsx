import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import travela from '../../images/travela.svg';
import icon from '../../images/drop-down-icon.svg';
import notification from '../../images/notification.svg';
import SearchBar from '../search-bar/SearchBar';
import Button from '../buttons/Buttons';
import ImageLink from '../image-link/ImageLink';
import { logoutUser } from '../../helper/userDetails';
import './NavBar.scss';

/**
 * @description - Contains SearhBar componentents, Logo, user avatar and logout
 *
 * @class NavBar
 *
 * @extends {PureComponent}
 *
 */

export class NavBar extends PureComponent {
  logout = () => {
    const { history } = this.props;
    logoutUser(history);
  };

  logoutLink() {
    return (
      <span>
        <Button
          imageSrc={icon}
          altText="Dropdown Icon"
          buttonId="demo-menu-lower-right"
          imageClass="navbar__mdl-Icon"
          buttonType="button"
          buttonClass="mdl-button mdl-js-button mdl-button--icon mdl-Icons"
        />
        <div className="navbar__mdl-list">
          <ul
            htmlFor="demo-menu-lower-right"
            className="mdl-menu mdl-menu--bottom-right mdl-js-menu mdl-js-ripple-effect navbar__menu"
          >
            <li className="mdl-menu__item navbar__menu-item">
              <div onClick={this.logout} id="logout" role="presentation">
                Logout
              </div>
            </li>
          </ul>
        </div>
      </span>
    );
  }

  renderNotification() {
    const { onNotificationToggle } = this.props;
    return (
      <div
        id="notification"
        onClick={onNotificationToggle}
        className="navbar__nav-size"
        role="presentation"
      >
        <span className="material-icons mdl-badge navbar__badge" data-badge="12">
          <img
            src={notification}
            alt="Notification"
            className="navbar__navbar-notification"
          />
        </span>
      </div>
    );
  }

  renderLogo() {
    return (
      <span className="navbar__logo-icons">
        <img src={travela} alt="Andela Logo" />
      </span>
    );
  }

  renderUserIcons() {
    const { avatar, user } = this.props;
    return (
      <div>
        <span className="navbar__mdl-icons">
          <ImageLink
            imageSrc={user ? user.UserInfo.picture : avatar}
            altText="Andela Logo"
            imageClass="navbar__mdl-upic"
          />
          <span className="navbar__text-size">
            {user ? user.UserInfo.name : ''}
          </span>
        </span>
        {this.logoutLink()}
      </div>
    );
  }

  renderHeader(){
    return(
      <div className="mdl-layout__header-row">
        <div className="navbar__nav-size navbar__logo-icons">
          {this.renderLogo()}
        </div>
        <div className="mdl-layout-spacer" />
        <div className="navbar__search-size mdl-cell--hide-phone">
          <SearchBar />
        </div>
        <nav className="mdl-navigation">
          {this.renderNotification()}
          <div className="navbar__user-icon navbar__nav-size mdl-cell--hide-tablet mdl-cell--hide-phone">
            {this.renderUserIcons()}
          </div>
        </nav>
      </div>
    );
  }

  render() {
    const {handleHideSearchBar, openSearch } = this.props;
    let showSearch='none';
    if(openSearch) {
      showSearch='block';
    }
    return (
      <header className="mdl-layout__header navbar__layout_header">
        {this.renderHeader()}
        <button type="button" className="navbar__search-icon--btn" onClick={handleHideSearchBar}>
          <div>
            <i className="material-icons navbar__search-icon">
          search
            </i>
          </div>
        </button>
        <div className="navbar__search-size mdl-cell--hide-desktop mdl-cell--hide-tablet" style={{display: `${showSearch}`}}>
          <SearchBar />
        </div>
      </header>
    );
  }
}

NavBar.propTypes = {
  onNotificationToggle: PropTypes.func.isRequired,
  history: PropTypes.shape({}).isRequired,
  user: PropTypes.shape({}).isRequired,
  avatar: PropTypes.string.isRequired,
  handleHideSearchBar: PropTypes.func.isRequired,
  openSearch: PropTypes.bool,
};

NavBar.defaultProps = {
  openSearch: false
};

const mapStateToProps = state => {
  return {
    user: state.auth.user
  };
};

export default withRouter(connect(mapStateToProps)(NavBar));
