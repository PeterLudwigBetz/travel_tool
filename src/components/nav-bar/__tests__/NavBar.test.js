import React from 'react';
import Cookie from 'cookies-js';
import { shallow } from 'enzyme';
import { NavBar } from '../NavBar';

// describe what we are testing
describe('Render NavBar component', () => {
  const props = {
    onNotificationToggle: jest.fn(),
    avatar: 'avatar',
    history: {
      push: jest.fn()
    },
    openSearch: true,
    handleHideSearchBar: jest.fn(),
    user: {
      UserInfo: {
        name: 'Tomato Jos',
        picture: 'http://picture.com/gif'
      }
    },
  };
  // make our assertions and what we expect to happen
  it('should match snapshot', () => {
    const wrapper = shallow(<NavBar {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the navbar as expected', () => {
    const wrapper = shallow(<NavBar {...props} />);
    expect(wrapper.length).toBe(1);
  });

  it('should render the OnNotification click as exepected', () => {
    const wrapper = shallow(<NavBar {...props} />);
    let NotificationToggleSpy = jest.spyOn(
      wrapper.instance().props,
      'onNotificationToggle'
    );
    wrapper
      .find('#notification')
      .first()
      .simulate('click');
    expect(NotificationToggleSpy).toHaveBeenCalled();
  });

  it('should log user out when the logout link is clicked', () => {
    const wrapper = shallow(<NavBar {...props} />);
    wrapper.find('#logout').simulate('click');
    const token = Cookie.get('login-status');
    const loginStatus = Cookie.get('jwt-token');
    expect(token).toEqual(undefined);
    expect(loginStatus).toEqual(undefined);
  });

  it('should should be search bar for phone view toggle display', () => {
    const wrapper = shallow(<NavBar {...props} />);
    expect(wrapper.length).toBe(1);
  });

  it('should log user out when the logout link is clicked', () => {
    const wrapper = shallow(<NavBar {...props} />);
    wrapper.find('#logout').simulate('click');
    const token = Cookie.get('login-status');
    const loginStatus = Cookie.get('jwt-token');
    expect(token).toEqual(undefined);
    expect(loginStatus).toEqual(undefined);
  })
});
