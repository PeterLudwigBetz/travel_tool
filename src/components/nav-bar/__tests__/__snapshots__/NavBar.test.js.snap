// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render NavBar component should match snapshot 1`] = `
<header
  className="mdl-layout__header navbar__layout_header"
>
  <div
    className="mdl-layout__header-row"
  >
    <div
      className="navbar__nav-size navbar__logo-icons"
    >
      <span
        className="navbar__logo-icons"
      >
        <img
          alt="Andela Logo"
          src="travela.svg"
        />
      </span>
    </div>
    <div
      className="mdl-layout-spacer"
    />
    <div
      className="navbar__search-size mdl-cell--hide-phone"
    >
      <SearchBar />
    </div>
    <nav
      className="mdl-navigation"
    >
      <div
        className="navbar__nav-size"
        id="notification"
        onClick={[Function]}
        role="presentation"
      >
        <span
          className="material-icons mdl-badge navbar__badge"
          data-badge="12"
        >
          <img
            alt="Notification"
            className="navbar__navbar-notification"
            src="notification.svg"
          />
        </span>
      </div>
      <div
        className="navbar__user-icon navbar__nav-size mdl-cell--hide-tablet mdl-cell--hide-phone"
      >
        <div>
          <span
            className="navbar__mdl-icons"
          >
            <ImageLink
              altText="Andela Logo"
              imageClass="navbar__mdl-upic"
              imageSrc="avatar"
            />
            <span
              className="navbar__text-size"
            />
          </span>
          <span>
            <Button
              altText="Dropdown Icon"
              buttonClass="mdl-button mdl-js-button mdl-button--icon mdl-Icons"
              buttonId="demo-menu-lower-right"
              buttonType="button"
              imageClass="navbar__mdl-Icon"
              imageSrc="drop-down-icon.svg"
              onClick={null}
              text="Login to Get Started"
              textClass=""
            />
            <div
              className="navbar__mdl-list"
            >
              <ul
                className="mdl-menu mdl-menu--bottom-right mdl-js-menu mdl-js-ripple-effect navbar__menu"
                htmlFor="demo-menu-lower-right"
              >
                <li
                  className="mdl-menu__item navbar__menu-item"
                >
                  <div
                    id="logout"
                    onClick={[Function]}
                    role="presentation"
                  >
                    Logout
                  </div>
                </li>
              </ul>
            </div>
          </span>
        </div>
      </div>
    </nav>
  </div>
  <button
    onClick={[Function]}
    type="button"
  >
    <i
      className="material-icons navbar__search-icon"
    >
      search
    </i>
  </button>
  <div
    className="navbar__search-size mdl-cell--hide-desktop mdl-cell--hide-tablet"
    style={
      Object {
        "display": "block",
      }
    }
  >
    <SearchBar />
  </div>
</header>
`;
